src/classes/RD2_OpportunityEvaluationService.cls

```diff
@@ -280,20 +367,330 @@ public inherited sharing class RD2_OpportunityEvaluationService {
             'PaymentMethod__c, ' +
             'StartDate__c, ' +
             'EndDate__c ' +
-            'FROM RecurringDonationSchedules__r ' +
-            'WHERE Active__c = true)';
+            'FROM RecurringDonationSchedules__r)';
 
         queryFields.add(parentAccountField);
         queryFields.add(oppSubquery);
         queryFields.add(scheduleSubquery);
 
+        String soql = new UTIL_Query()
+        .withFrom(npe03__Recurring_Donation__c.SObjectType)
+        .withSelectFields(queryFields)
+        .withWhere('Id IN :rdIds')
+        .build();
+
+        return Database.query(soql);
+    }
+
+    /***
+    * @description valuate The RDs against the Installment Opp Auto Creation Option Setting
+    * @param rds: The list of recurring donation to be filtered
+    */
+    private void filterAgainstInstallmentAutoCreationOption(List<npe03__Recurring_Donation__c> rds) {
```

Comment:
The `needNextInstallment(rd)` and `rd.isNew()` should answer questions about the first/next installment Opp creation. So the query is more beneficial to occur before. Otherwise, it seems duplication of the decision making about the Opp creation and methods seem complex. Let me think about alternatives (if any).